src/System.IO.Compression/src/System/IO/Compression/DeflaterZLib.cs

```diff
@@ -35,48 +35,34 @@ internal DeflaterZLib()
         internal DeflaterZLib(CompressionLevel compressionLevel)
         {
             ZLibNative.CompressionLevel zlibCompressionLevel;
-            int windowBits;
             int memLevel;
-            ZLibNative.CompressionStrategy strategy;
 
             switch (compressionLevel)
             {
-                // Note that ZLib currently exactly correspond to the optimal values.
-                // However, we have determined the optimal values by intependent measurements across
-                // a range of all possible ZLib parameters and over a set of different data.
-                // We stress that by using explicitly the values obtained by the measurements rather than
-                // ZLib defaults even if they happened to be the same.
-                // For ZLib 1.2.3 we have (copied from ZLibNative.cs):
-                // ZLibNative.CompressionLevel.DefaultCompression = 6;
-                // ZLibNative.Deflate_DefaultWindowBits = -15;
-                // ZLibNative.Deflate_DefaultMemLevel = 8;
+                // See the note in ZLibNative.CompressionLevel for the recommended combinations.
 
                 case CompressionLevel.Optimal:
-                    zlibCompressionLevel = (ZLibNative.CompressionLevel)6;
-                    windowBits = -15;
-                    memLevel = 8;
-                    strategy = ZLibNative.CompressionStrategy.DefaultStrategy;
+                    zlibCompressionLevel = ZLibNative.CompressionLevel.BestCompression;
+                    memLevel = ZLibNative.Deflate_DefaultMemLevel;
                     break;
 
                 case CompressionLevel.Fastest:
                     zlibCompressionLevel = ZLibNative.CompressionLevel.BestSpeed;
-                    windowBits = -15;
-                    memLevel = 8;
-                    strategy = ZLibNative.CompressionStrategy.DefaultStrategy;
+                    memLevel = ZLibNative.Deflate_DefaultMemLevel;
                     break;
 
                 case CompressionLevel.NoCompression:
                     zlibCompressionLevel = ZLibNative.CompressionLevel.NoCompression;
-                    windowBits = -15;
                     memLevel = 7;
-                    strategy = ZLibNative.CompressionStrategy.DefaultStrategy;
```

Comment:
What does memLevel 7 mean? Can that be in a constant somewhere, if it's necessary? Is it different from the default level?